### PR TITLE
DOC: Add paper for JOSS submission

### PR DESCRIPTION
**Pull request by richford**
This PR adds a paper directory with a draft JOSS submission and updates some corresponding sections of the documentation. It also adds Jason and Noah as co-authors.

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    index: 4
+  - name: Graduate School of Education and Division of Developmental and Behavioral Pediatrics, Stanford University
+    index: 5
+date: 25 Dec 2021
+bibliography: paper.bib
+---
+
+## Summary
+
+For high-dimensional supervised learning, it is often benefitial to use
```
Comment (arokem) [inline paper/paper.md:41]:
```suggestion
For high-dimensional supervised learning, it is often beneficial to use
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+domain-specific knowledge to improve the performance of statistical learning
+models. When the problem contains covariates which form groups, researchers
+can including this grouping information to find parsimonious representations
```
Comment (arokem) [inline paper/paper.md:44]:
```suggestion
can include this grouping information to find parsimonious representations
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+examples of both strategies (grid search for a classification example and
+SMBO for a regression example) in the online documentation.
+
+## Acknowledgments
+
+Groupyr development is supported through a grant from the Gordon and Betty
+Moore Foundation and from the Alfred P. Sloan Foundation to the University of
+Washington eScience Institute, as well as NIMH BRAIN Initiative grant
+1RF1MH121868-01 to Ariel Rokem at the University of Washington. We would like
+to thank Noah Simon and Jason Yeatman for helpful discussions of the SGL
```
Comment (arokem) [inline paper/paper.md:156]: I think that we don't need to thank co-authors 😄

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+previously attempted hyperparameter combinations should provide the initial
+guess for the current evaluation. However, even without warm-starting, we
+find that the SMBO strategy usually outperforms grid search because far fewer
+evaluations are needed to arrive at the optimal hyperparameters. We provide
+examples of both strategies (grid search for a classification example and
+SMBO for a regression example) in the online documentation.
+
+## Acknowledgments
+
+Groupyr development is supported through a grant from the Gordon and Betty
```
Comment (arokem) [inline paper/paper.md:152]:
```suggestion
Groupyr development was supported through a grant from the Gordon and Betty
```

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+Moore Foundation and from the Alfred P. Sloan Foundation to the University of
+Washington eScience Institute, as well as NIMH BRAIN Initiative grant
+1RF1MH121868-01 to Ariel Rokem at the University of Washington. We would like
```
Comment (arokem) [inline paper/paper.md:155]:
```suggestion
1RF1MH121868-01 to Ariel Rokem at the University of Washington and through cloud credits from the Google Cloud Platform. We would like
```
Why not.

**Review by arokem — APPROVED** [review]
Looks great! A couple of small suggestions/comments.

File: paper/paper.md (hunk 3 above)
Comment (richford) [reply paper/paper.md]: Hah, thanks! Corrected in latest commit.

**Comment by richford** [comment]
Leaving this in WIP status until we get feedback (or lack thereof) from Noah, Jason, and Manjari.

**Pull request merged**